### ISSUE TITLE
GribEnvironment::DefinitionsPath must be a valid path

### ISSUE THREAD

**Issue by pavlose** [0x1mason/GribApi.NET#32]
Hi Eric,
I have an ASP.NET website (working with Visual Studio 2013) and I am trying to use GribApi.NET. I am trying to create a new GribFile, i.e.,

using(GribFile file = new GribFile(weatherpath)) { }

but I keep getting a GribApiException:

GribEnvironment::DefinitionsPath must be a valid path. If you're using ASP.NET or NUnit, this exception is usually caused by shadow copying. Please see GribApi.NET's documentation for help. Path:... 

I used your suggestion for shadow copying, i.e., setting

Environment.SetEnvironmentVariable("GRIB_API_DIR_ROOT", "C:\\Some\\Path", EnvironmentVariableTarget.Process);

but it did not work. I tried to set GribEnvironment.DefinitionsPath myself but it did not work (same error message). Moreover, while debugging, I noticed that GribEnvironment.GribApiVersion has the indication:

"The type initializer for 'Grib.Api.Interop.SWIG.GribApiProxy' threw an exception."

 Do you have any suggestions?

**Comment by 0x1mason** [comment]
I moved recently and do not have my windows machine set up yet. A few ideas in the interim:
- At the entry point to you application try changing the working directory to the location of the GRIB.api.dll. Make sure it has the Grib.Api directory in it. Then call GribEnvironment.Init().
- Copy the GRIB.Api dir to your working directory. Then call GribEnv.Init. Try the same thing with the definitions directory.
- What if you hard code the env var GRIB_DEFINITION_PATH or set it in the build/debug properties? The idea is to make it available as soon as the app starts.
- Try setting the definitions path as a relative path. To _dll_ as opposed to your app. Use `/` instead of `\`
- Add the folder containing the Grib.Api directory to PATH
- Add the the folder containing the definitions directory (which is in the Grib.Api directory) to PATH
- Uber hacky and for testing purposes only, copy the definitions folder to a well-known location such as Windows/System32 folder.

Take a look at GribEnvironment.cs and GribEnvLoadHelper.cs to get an idea of what's going on.

I'll try to get my machine setup soon so I can give you move concrete help.

**Comment by pavlose** [comment]
Hi Eric, 
Still not able to make it work. 
Meanwhile, I tried it in a non-ASP.NET app and it worked perfectly.
I will let you know if I have any progress.
In any case, many thanks for your help.

**Comment by 0x1mason** [comment]
K, will try to get to it soon. I know I made it work at some point.

**Comment by 0x1mason** [comment]
@pavlose 
Sorry for the delay. This works for me:

``` csharp
Environment.SetEnvironmentVariable("GRIB_API_DIR_ROOT", AppDomain.CurrentDomain.RelativeSearchPath);
```

I'll fix this in the upcoming release so that you don't have to set this manually.

**Comment by 0x1mason** [comment]
Note, n my ASP.NET project, the Grib.Net reference, "Copy Local" is set to true.

**Comment by 0x1mason** [comment]
Please let me know if that works.

I hope to have the fix available via nuget in the next 48 hours.

**Comment by 0x1mason** [comment]
0.7.1 is available on nuget.

**Comment by pavlose** [comment]
Sorry for taking so long to get back to you.
I got the newer version 0.7.1.  Unfortunately it does not resolve my issue. I tried setting "Copy local" to true, but there is no such option for Grib.api in the reference's properties. I am not sure why this is so...
Meanwhile, I have set up a (non-ASP.net) project and started working with Grib.Api. There, "Copy local" is actually set to true, and Grib.api works like a charm (many thanks!!)

**Comment by 0x1mason** [comment]
Try this: https://github.com/0x1mason/GribApiAspNet

You'll need to restore all the packages via the VS Nuget UI or via the Pkg Manager Console:

``` shell
PM> nuget restore
```

**Comment by 0x1mason** [comment]
The actual code is in `Default.aspx.cs`.

**Comment by pavlose** [comment]
GribApiAspNet worked. I also created a new Project->ASP.NET Web Application and it worked.
It didn't work in my previous tests, because I had created (and was testing it via) a Website->ASP.NET Empty Website. This is also why I saw no "References" in the solution explorer. Probably because of the differences between asp.net web applications and websites in the solution structure and the way they handle references and dlls.
